src/code/z_actor.c

```diff
@@ -841,54 +841,73 @@ void Actor_Destroy(Actor* actor, PlayState* play) {
     }
 }
 
-void func_8002D7EC(Actor* actor) {
+/**
+ * Update actor's position factoring in velocity and collider displacement
+ */
+void Actor_UpdatePos(Actor* actor) {
     f32 speedRate = R_UPDATE_RATE * 0.5f;
 
     actor->world.pos.x += (actor->velocity.x * speedRate) + actor->colChkInfo.displacement.x;
     actor->world.pos.y += (actor->velocity.y * speedRate) + actor->colChkInfo.displacement.y;
     actor->world.pos.z += (actor->velocity.z * speedRate) + actor->colChkInfo.displacement.z;
 }
 
-void func_8002D868(Actor* actor) {
+/**
+ * Update actor's velocity accounting for gravity (without exceeding terminal velocity)
+ */
+void Actor_UpdateVelocityWithGravity(Actor* actor) {
     actor->velocity.x = Math_SinS(actor->world.rot.y) * actor->speedXZ;
     actor->velocity.z = Math_CosS(actor->world.rot.y) * actor->speedXZ;
 
     actor->velocity.y += actor->gravity;
+
     if (actor->velocity.y < actor->minVelocityY) {
         actor->velocity.y = actor->minVelocityY;
     }
 }
 
-void Actor_MoveForward(Actor* actor) {
-    func_8002D868(actor);
-    func_8002D7EC(actor);
+/**
+ * Move actor while accounting for its current velocity and gravity.
+ * The actor will move in the direction of its world yaw.
+ */
+void Actor_MoveWithGravity(Actor* actor) {
+    Actor_UpdateVelocityWithGravity(actor);
+    Actor_UpdatePos(actor);
 }
 
-void func_8002D908(Actor* actor) {
+/**
+ * Update actor's velocity without gravity.
+ */
+void Actor_UpdateVelocityWithoutGravity(Actor* actor) {
     f32 sp24 = Math_CosS(actor->world.rot.x) * actor->speedXZ;
 
     actor->velocity.x = Math_SinS(actor->world.rot.y) * sp24;
     actor->velocity.y = Math_SinS(actor->world.rot.x) * actor->speedXZ;
     actor->velocity.z = Math_CosS(actor->world.rot.y) * sp24;
```

Comment:
any ideas for this temp name, lmk. `xzSpeedScale`? should it involve pitch as well? idk

Reply:
So `Actor_UpdateVelocityWithoutGravity` accounts for pitch but `Actor_UpdateVelocityWithGravity` does not. I'm just trying to compare that to `func_8002D9A4` which also does account for pitch. `Actor_UpdateSpeedWithoutGravity`?

For `sp24`, camera would name that `pitchAdjSpeedXZ` (pitch adjusted speed xz) as an idea

Reply:
uh oh, does the instance variable need to change? lmao. since its context dependent on what move function is used..
speedXZ/the xz + gravity function is way more common, and is why it has the name it does. but... yeah

Reply:
more I think about it, the instance variable probably should be just `speed`, with a comment clarifying that it can be XZ or XYZ depending on which move function is used

Reply:
I thought we were aware speedXZ not always being speedXZ and being fine with the """odd""" uses (at least I was aware :sweat_smile: )

If that's not the case yeah it may be worth debating renaming it to `speed`

Reply:
I think i instinctively knew cause Id still set it when using the non-gravity move functions, but seeing it pointed out like this is still surprising to me yea

I think its just called speedXZ due to legacy reasons (rom hackers + speedrunners calling it "linear velocity" wrt link)

Reply:
also with this change it would make more sense to include which axis youre moving on in the name of the function call, since it literally does depend on that and the speed value tells you nothing about it. So in that case I would deal with the "clunky" names.

Reply:
For context, #1477 happened (Actor.speedXZ -> speed) and this function is as of writing this `Actor_UpdateVelocityXYZ` with the temp named `speedXZ`